src/hotspot/share/gc/shared/partialArraySplitter.inline.hpp

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#ifndef SHARE_GC_SHARED_PARTIALARRAYSPLITTER_INLINE_HPP
+#define SHARE_GC_SHARED_PARTIALARRAYSPLITTER_INLINE_HPP
+
+#include "gc/shared/partialArraySplitter.hpp"
+
+#include "gc/shared/partialArrayTaskStats.hpp"
+#include "gc/shared/partialArrayTaskStepper.inline.hpp"
+#include "gc/shared/taskqueue.inline.hpp"
+#include "oops/oop.hpp"
+#include "utilities/debug.hpp"
+#include "utilities/globalDefinitions.hpp"
+#include "utilities/macros.hpp"
+
+template<typename Queue>
+size_t PartialArraySplitter::start(Queue* queue,
+                                   objArrayOop source,
+                                   objArrayOop destination,
+                                   size_t length) {
+  PartialArrayTaskStepper::Step step = _stepper.start(length);
+  // Push any needed partial scan tasks.  Pushed before processing the initial
+  // chunk to allow other workers to steal while we're processing.
+  if (step._ncreate > 0) {
+    TASKQUEUE_STATS_ONLY(_stats.inc_split(););
+    TASKQUEUE_STATS_ONLY(_stats.inc_pushed(step._ncreate);)
+    PartialArrayState* state =
+      _allocator.allocate(source, destination, step._index, length, step._ncreate);
+    for (uint i = 0; i < step._ncreate; ++i) {
+      queue->push(ScannerTask(state));
+    }
+  } else {
+    assert(step._index == length, "invariant");
+  }
+  return step._index;
+}
+
+template<typename Queue>
+PartialArraySplitter::Step
+PartialArraySplitter::step(PartialArrayState* state, Queue* queue, bool stolen) {
```

Comment:
Probably easier to read if we rename to `claim`,  step is used as noun in many other places

Reply:
I like the suggested name change.